src/airobot/robot/ur5e_pybullet.py

```diff
@@ -0,0 +1,721 @@
+                jnt_diff = new_jnt_poss - tgt_jnt_poss
+                error = np.max(np.abs(jnt_diff))
+                if error < self.cfgs.MAX_JOINT_ERROR:
+                    break
+                time.sleep(0.001)
+        return success
+
+    def enable_torque_control(self, joint_name=None):
+        """
+        Enable the torque control mode in Pybullet
+
+        Args:
+            joint_name: If it's none, then all the six joints
+                on the UR robot are enabled in torque control mode.
+                Otherwise, only the specified joint is enabled
+                in torque control mode.
+
+        """
+        if joint_name is None:
+            tgt_vels = [0.0] * len(self.actuator_ids[:-1])
+            forces = [0.0] * len(self.actuator_ids[:-1])
+            p.setJointMotorControlArray(self.robot_id,
+                                        self.actuator_ids[:-1],
+                                        p.VELOCITY_CONTROL,
+                                        targetVelocities=tgt_vels,
+                                        forces=forces)
+        else:
+            jnt_id = self.jnt_to_id[joint_name]
+            p.setJointMotorControl2(self.robot_id,
+                                    jnt_id,
+                                    p.VELOCITY_CONTROL,
+                                    targetVelocity=0,
+                                    force=0.0)
+
+    def disable_torque_control(self, joint_name=None):
+        """
+        Disable the torque control mode in Pybullet
+
+        Args:
+            joint_name: If it's none, then all the six joints
+                on the UR robot are disabled with torque control.
+                Otherwise, only the specified joint is disabled with
+                torque control.
+                The joint(s) will enter velocity control mode.
+
+        """
+        if joint_name is None:
+            self.set_jvel([0.0] * 7)
+        else:
+            self.set_jvel(0.0, joint_name)
+
+    def get_jpos(self, joint_name=None):
+        """
+        Return the joint position(s)
+
+        Args:
+            joint_name: If it's None, it will return joint positions
+                of all the actuated joints. Otherwise, it will
+                return the joint position of the specified joint
+
+        Returns:
+            joint position (float) or joint positions (list) depends on
+            the joint_name
+        """
+        if joint_name is None:
+            states = p.getJointStates(self.robot_id, self.actuator_ids)
+            pos = [state[0] for state in states]
+        else:
+            jnt_id = self.jnt_to_id[joint_name]
+            pos = p.getJointState(self.robot_id, jnt_id)[0]
+        return pos
+
+    def get_jvel(self, joint_name=None):
+        """
+        Return the joint velocity(ies)
+
+        Args:
+            joint_name: If it's None, it will return joint velocities
+                of all the actuated joints. Otherwise, it will
```

Comment:
all actuated joints -- means all the joints? If so, remove the word "actuated".

Reply:
Nope. There are some joints on the fingers that are underactuated. There is only one actuated joint on the gripper.